src/python/pants/build_graph/address.py

```diff
@@ -283,11 +283,12 @@ def to_address(self):
     return Address(spec_path=self.spec_path, target_name=self.target_name)
 
   @property
+  @deprecated('1.5.0.dev0',
+              hint_message='Use `BuildFileAddress.rel_path` to access the relative path to the '
+                           'BUILD file for a target.')
   def build_file(self):
     """The build file that contains the object this address points to.
 
-    :API: public
```

Comment:
should this stay marked `:API: public` until it's removal?

technically, it's still public - just deprecated - until it's removed entirely.